File: Backend/main-backend-app/reservations-service/src/main/java/rs/ac/uns/ftn/xml/team17/reservationsservice/dto/reservation/ReservationPreview.java
package rs.ac.uns.ftn.xml.team17.reservationsservice.dto.reservation;

import java.util.Date;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import rs.ac.uns.ftn.xml.team17.reservationsservice.model.address.Address;
import rs.ac.uns.ftn.xml.team17.reservationsservice.model.hotel.Hotel;
import rs.ac.uns.ftn.xml.team17.reservationsservice.model.reservation.Reservation.ReservationStatus;

@NoArgsConstructor
@Getter
@Setter
public class ReservationPreview {
	/*
	 * Detalji o rezervaciji
	 */
	private Integer id;
	private Date start;
	private Date end;
	private Double totalPrice;
	private ReservationStatus status;
	private Boolean canCancel;
	/*
	 * Detalji o hotelu
	 */
	private Hotel hotel;
	/*
	 * Detalji o sobi
	 */
	private Integer roomId;
	private Address address;
	private Integer roomNumber;
	private Integer numberOfPeople;
}

File: Frontend/client-front/src/app/model/reservation/reservationPreview.ts
import { HotelPreview } from '../hotel/hotelPreview';
import { Address } from '../address';

export class ReservationPreview {

    // Details about reservation
    id: number;
    created: Date = new Date();
    start: Date = new Date();
    end: Date = new Date();
    totalPrice: number;
    status: string;
    canCancel: boolean;

    // Details about hotel
    hotel: HotelPreview = new HotelPreview();

    // Details about room
    roomId: number;
    address: Address = new Address();
    roomNumber: number;
    // floorNumber: number;
    numberOfPeople: number;
}
